archiver.go

```diff
@@ -65,6 +65,7 @@ type Archiver interface {
 	// are walked and recursively added, preserving folder
 	// structure.
 	Archive(sources []string, destination string) error
+	ArchiveToStream(output io.Writer, sources []string) error
```

Comment:
Instead of modifying existing interfaces, let's make new ones: `StreamArchiver` and `StreamUnarchiver`, maybe. And, although this is yak-shaving, let's reverse the order of the arguments to match Archive and Unarchive; in, out; not out, in.